pkg/filter/network/proxy/factory.go

```diff
@@ -79,7 +79,15 @@ func CreateProxyFactory(conf map[string]interface{}) (api.NetworkFilterChainFact
 	}
 
 	if len(p.ExtendConfig) != 0 {
-		gfcf.extendConfig = protocol.HandleConfig(api.ProtocolName(p.DownstreamProtocol), p.ExtendConfig)
+		gfcf.extendConfig = make(map[api.ProtocolName]interface{})
+		if len(gfcf.protocols) == 1 && gfcf.protocols[0] != protocol.Auto {
+			proto := gfcf.protocols[0]
+			gfcf.extendConfig[proto] = protocol.HandleConfig(proto, p.ExtendConfig)
+		} else {
+			for proto, _ := range p.ExtendConfig {
+				gfcf.extendConfig[api.ProtocolName(proto)] = protocol.HandleConfig(api.ProtocolName(proto), p.ExtendConfig[proto])
```

Comment:
Maybe it's better that use the value of `proto,value := range p.ExtendConfig` instead of `p.ExtendConfig[proto]`.